dll/shellext/zipfldr/CZipCreator.cpp

```diff
@@ -279,6 +275,7 @@ unsigned CZipCreatorImpl::JustDoIt()
 
     int err = 0;
     CStringW strTarget, strBaseName = DoGetBaseName(m_items[0]);
+    const BOOL bUtf8 = TRUE;
```

Comment:
Why not then just use directly "TRUE" whenever this is used?

Reply:
In the future, some customers might want to use non-UTF-8. Just a memorial reason.

Reply:
I'd suggest to loop through all files to be packed first, check if their names contain non-latin (non-ANSI) characters, and then enable UTF-8 only in this case.

Reply:
A Latin character in the different codepage also causes mojibake.
A plain ASCII string has the same size of the UTF-8 string of it.

Reply:
A Latin character in the different codepage also causes mojibake.
A plain ASCII string has the same size of the UTF-8 string.

Reply:
I wonder how these UTF-8 zip files would look like in Windows XP/2003 zip folder.

Reply:
You can check UTF-8 Zip Folders:
[新建文件夹.zip](https://github.com/reactos/reactos/files/12105683/default.zip)
[新しいフォルダ.zip](https://github.com/reactos/reactos/files/12105684/default.zip)
* Chinese and Japanese fonts required.

Reply:
To avoid complexity with fonts, I tried this scenario:

- Create zip-file from ReactOS, with non-ANSI filenames (I used Cyrillic chars here)
- Open the same zip-file in Windows XP/2003, result is mojibake:

| ReactOS | Windows XP |
| --- | --- |
| ![image](https://github.com/reactos/reactos/assets/578406/5b8de8e7-e265-442e-98ea-9b8ad3df3199) | ![image](https://github.com/reactos/reactos/assets/578406/4e9c95d2-c7d1-4453-ae26-04ce823c4f2e) |

So the problem that I expected does happen. This supports the idea of registry key you had added earlier here.

Reply:
Now the registry values are effective. Fixed.

Reply:
Now codepage is user selectable.